supertokens/supertokens.go

```diff
@@ -119,44 +122,63 @@ func (s *SuperTokens) SendTelemetry() {
 
 func (s *SuperTokens) Middleware() http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		reqURL, _ := NewNormalisedURLPath(r.URL.Path) // todo: error handle
+		reqURL, _ := NewNormalisedURLPath(r.RemoteAddr) // todo: error handle
 		path := s.AppInfo.APIGatewayPath.AppendPath(*reqURL)
-		// method := normaliseHttpMethod(r.Method)
+		method := r.Method
 
-		if strings.HasPrefix(path.GetAsStringDangerous(), s.AppInfo.APIBasePath.GetAsStringDangerous()) == false {
+		if !strings.HasPrefix(path.GetAsStringDangerous(), s.AppInfo.APIBasePath.GetAsStringDangerous()) {
 			return
 		}
 		requestRID := getRIDFromRequest(r)
 		if requestRID != "" {
-			var matchedRecipe RecipeModule
+			var matchedRecipe *RecipeModule
 			for _, recipeModule := range s.RecipeModules {
 				if recipeModule.GetRecipeID() == requestRID {
-					matchedRecipe = recipeModule
+					matchedRecipe = &recipeModule
 					break
 				}
 			}
-			if reflect.DeepEqual(matchedRecipe, RecipeModule{}) {
+			if matchedRecipe == nil {
 				return
 			}
 
+			id := matchedRecipe.ReturnAPIIdIfCanHandleRequest(path, method)
+			if id == "" {
+				return
+			}
+			s.HandleAPI(*matchedRecipe, id, r, w, path, method)
```

Comment:
These can throw an error right (same for the one below)? How will those be handled?